sympy/simplify/precision_optimize.py

```diff
@@ -0,0 +1,8 @@
+from sympy import simplify
+
+def algebraic_rewrite(expr):
+    rewritten = expr.rewrite()
+    simplified = simplify(rewritten)
+    if simplified != expr:
+        print(f"Rewritten {expr} -> {simplified}")
+    return simplified
```

Comment:
This should not call `simplify` which is a function that should not be used in library code.